libraries/AC_AttitudeControl/AC_WeatherVane.cpp

```diff
@@ -0,0 +1,229 @@
+    // @Range: 0 10
+    // @Increment: 0.1
+    // @User: Standard
+    AP_GROUPINFO("ANG_MIN", 3, AC_WeatherVane, _min_dz_ang_deg, 1.0),
+
+    // @Param: HGT_MIN
+    // @DisplayName: Weathervaning min height
+    // @Description{Copter}: Above this height weathervaning is permitted.  If a range finder is fitted or if terrain is enabled, this parameter sets height AGL.  Otherwise, this parameter sets height above home.  Set zero to ignore minimum height requirement to activate weathervaning.
+    // @Description{Plane}: Above this height weathervaning is permitted.  If RNGFND_LANDING is enabled or terrain is enabled then this parameter sets height AGL. Otherwise this parameter sets height above home.  Set zero to ignore minimum height requirement to activate weathervaning
+    // @Units: m
+    // @Range: 0 50
+    // @Increment: 1
+    // @User: Standard
+    AP_GROUPINFO("HGT_MIN", 4, AC_WeatherVane, _min_height, 0.0),
+
+    // @Param: SPD_MAX
+    // @DisplayName: Weathervaning max ground speed
+    // @Description: Below this ground speed weathervaning is permitted. Set to 0 to ignore this condition when checking if vehicle should weathervane.
+    // @Units: m/s
+    // @Range: 0 50
+    // @Increment: 0.1
+    // @User: Standard
+    AP_GROUPINFO("SPD_MAX", 5, AC_WeatherVane, _max_vel_xy, WVANE_PARAM_SPD_MAX_DEFAULT),
+
+    // @Param: VELZ_MAX
+    // @DisplayName: Weathervaning max vertical speed
+    // @Description: The maximum climb or descent speed that the vehicle will still attempt to weathervane. Set to 0 to ignore this condition to get the aircraft to weathervane at any climb/descent rate.  This is particularly useful for aircraft with low disc loading that struggle with yaw control in decent.
+    // @Units: m/s
+    // @Range: 0 5
+    // @Increment: 0.1
+    // @User: Standard
+    AP_GROUPINFO("VELZ_MAX", 6, AC_WeatherVane, _max_vel_z, WVANE_PARAM_VELZ_MAX_DEFAULT),
+
+    // @Param: TAKEOFF
+    // @DisplayName: Takeoff override
+    // @Description: Override the weather vaning behaviour when in takeoffs
+    // @Values: -1:No override,0:Disabled,1:Nose into wind,2:Nose or tail into wind,3:Side into wind,4:tail into wind
+    // @User: Standard
+    AP_GROUPINFO("TAKEOFF", 7, AC_WeatherVane, _takeoff_direction, -1),
+
+    // @Param: LAND
+    // @DisplayName: Landing override
+    // @Description: Override the weather vaning behaviour when in landing
+    // @Values: -1:No override,0:Disabled,1:Nose into wind,2:Nose or tail into wind,3:Side into wind,4:tail into wind
+    // @User: Standard
+    AP_GROUPINFO("LAND", 8, AC_WeatherVane, _landing_direction, -1),
+
+    AP_GROUPEND
+};
+
+
+// Constructor
+AC_WeatherVane::AC_WeatherVane(void)
+{
+    AP_Param::setup_object_defaults(this, var_info);
+    _singleton = this;
+}
+
+bool AC_WeatherVane::get_yaw_rate_cds(float &yaw_output, const int16_t pilot_yaw, const float hgt, const float roll_cdeg, const float pitch_cdeg, const bool is_takeoff, const bool is_landing)
```

Comment:
```suggestion
bool AC_WeatherVane::get_yaw_out(float &yaw_output, const int16_t pilot_yaw, const float hgt, const float roll_cdeg, const float pitch_cdeg, const bool is_takeoff, const bool is_landing)
```
Its not really a yaw rate anymore.